lib/SimpleSAML/Modules/Module.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+
+namespace SimpleSAML\Modules;
+
+
+use Symfony\Component\Config\Exception\FileLocatorFileNotFoundException;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Loader\DirectoryLoader;
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+abstract class Module extends Bundle
+{
+    abstract public function getShortName(): string;
+
+    public function build(ContainerBuilder $container)
```

Comment:
This method is called when the cache is warmed. Here we can define the services this module will use.